echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/orca/PipelineInitiator.java

```diff
@@ -166,7 +167,20 @@ public void startPipeline(Pipeline pipeline, TriggerSource triggerSource) {
 
   private void triggerPipeline(Pipeline pipeline, TriggerSource triggerSource)
       throws RejectedExecutionException {
-    executorService.submit(() -> triggerPipelineImpl(pipeline, triggerSource));
+    final Map<String, String> mdc = MDC.getCopyOfContextMap();
```

Comment:
Curious on your thought process behind this vs `propagate()`?

Reply:
Is there any sort of test we could add around this ...

Reply:
I added a test, but then it actually didn't work. The problem is that the test is using a directExecutor and then it actually calls from the same thread... I will add a test but i will just need more time to figure out how to block on that thread in spock...

`propagate()` doesn't actually carry the MDC across thread boundary, annoyingly - maybe that was the intent but i don't see how it would actually do that short of what I am doing

Reply:
`propagate()` should be able to propagate across threads, it's not going to blindly propagate all MDC but that shouldn't matter?

But you'd need to call `propagate()` outside the `submit()` and pass the resulting `Callable<>` into it.

My go to way of testing something like this would to do an assert within the mock ... roughly something like:

```
1 * orca.blah() { 
  assert MDC.get("blah") == "what I expect"
}
```

Reply:
Thanks for tip - added a test and used `propagate` - much better this way!